src/neuroglancer/viewer.ts

```diff
@@ -524,6 +536,47 @@ export class Viewer extends RefCounted implements ViewerState {
       topRow.appendChild(button);
     }
 
+    if (typeof STATE_SERVERS !== 'undefined') {
```

Comment:
Thanks for the changes.  This is looking pretty close.  Can you keep this in a separate file, though, since viewer.ts is already quite large?

Reply:
Yeah I'll do that.

I replaced the dialog with a very basic UI like you suggested.
![state-share-ui](https://user-images.githubusercontent.com/188155/115622172-821b0d00-a2c5-11eb-8313-90a46ce10c62.png)

The select element only comes up when there is more than one state server option.

It is controlled by a config file like
```json
{
    "test": {
        "url": "middleauth+https://authsl1.middleauth.com/nglstate/post",
    },
    "production": {
        "url": "middleauth+https://global.daf-apis.com/nglstate/post",
        "default": true
    }
}
```
I bundled it using `node ./config/esbuild-cli.js --config=min --no-typecheck --define STATE_SERVERS=$(cat config/state_servers.json | tr -d ” \t\n\r”)`

I had to update esbuild for that define to work in github actions. >= 0.9 seems to work

When you share, it copies it to clipboard.

You can test it out here: https://state-share-2-dot-neuroglancer-dot-seung-lab.appspot.com

Reply:
Thanks for these changes.  I've tried it out and it looks pretty good!

Reply:
Thanks, I moved the code back into a separate file. I kept the trackable value in view, not sure if it makes sense to move that into the StateShare object. I also wasn't sure about exporting a `stateShareEnabled` variable so that the viewer would know whether to instantiate the StateShare object.